test/server/controllers/UserController.test.js

```diff
@@ -50,3 +53,109 @@ describe('/users/signup', () => {
       });
   });
 });
+
+describe('Testing Login feature -Integration testng', () => {
+  it(
+    'should display error message if user logins with empty field',
+    (done) => {
+      chai.request(app)
+        .post(loginUrl)
+        .send({
+          email: '',
+          password: ''
+        })
+        .end((err, res) => {
+          res.status.should.be.eql(422);
+          res.body.should.be.eql({
+            errors: {
+              email: ['please enter email']
+            }
+          });
+        });
+      done();
+    }
+  );
+
+  it(
+    'should display error message if user logins with empty password',
+    (done) => {
+      chai.request(app)
+        .post(loginUrl)
+        .send({
+          email: 'augustineezinwa@gmail.com',
+          password: ''
+        })
+        .end((err, res) => {
+          res.status.should.be.eql(422);
+          res.body.should.be.eql({
+            errors: {
+              password: ['please enter password']
+            }
+          });
+        });
+      done();
+    }
+  );
+
+  it(
```

Comment:
Whats the difference between this test and the test on line 141?

Reply:
The test on line 80 is the scenario when the user logs in without providing a password. While the test on line 141 is the scenario where the user provides incorrect login credentials, be it password or email or both.

Reply:
`should display error message if user logs in with invalid credentials` and `should return error if user logs in with wrong password` are the tests I'm referring to.

Reply:
Ok.